tfe/data_source_outputs.go

```diff
@@ -0,0 +1,256 @@
+					Summary:  "Error encoding state",
+					Detail:   fmt.Sprintf("Error encoding state: %s", err.Error()),
+				},
+			},
+		}, nil
+	}
+	return &tfprotov5.ReadDataSourceResponse{
+		State: &state,
+	}, nil
+}
+
+func (d dataSourceOutputs) ValidateDataSourceConfig(ctx context.Context, req *tfprotov5.ValidateDataSourceConfigRequest) (*tfprotov5.ValidateDataSourceConfigResponse, error) {
+	return &tfprotov5.ValidateDataSourceConfigResponse{}, nil
+}
+
+func (d dataSourceOutputs) readConfigValues(req *tfprotov5.ReadDataSourceRequest) (string, string, error) {
+	var orgName string
+	var wsName string
+	var err error
+
+	config := req.Config
+	val, err := config.Unmarshal(tftypes.Object{
+		AttributeTypes: map[string]tftypes.Type{
+			"workspace":    tftypes.String,
+			"organization": tftypes.String,
+			"values":       tftypes.DynamicPseudoType,
+			"id":           tftypes.String,
+		}})
+	if err != nil {
+		return orgName, wsName, fmt.Errorf("Error unmarshalling config: %v", err)
+	}
+
+	var valMap map[string]tftypes.Value
+	err = val.As(&valMap)
+	if err != nil {
+		return orgName, wsName, fmt.Errorf("Error assigning configuration attributes to map: %v", err)
+	}
+
+	if valMap["organization"].IsNull() || valMap["workspace"].IsNull() {
+		return orgName, wsName, fmt.Errorf("Organization and Workspace cannot be nil: %v", err)
+	}
+
+	err = valMap["organization"].As(&orgName)
+	if err != nil {
+		return orgName, wsName, fmt.Errorf("Error assigning 'organization' value to string: %v", err)
+	}
+
+	err = valMap["workspace"].As(&wsName)
+	if err != nil {
+		return orgName, wsName, fmt.Errorf("Error assigning 'workspace' value to string: %v", err)
+	}
+
+	return orgName, wsName, nil
+}
+
+type rootModule struct {
+	RootOutputs map[string]rawOutput `json:"outputs"`
+}
+
+type rawOutput struct {
+	ValueRaw     json.RawMessage `json:"value"`
+	ValueTypeRaw json.RawMessage `json:"type"`
+}
+
+type stateData struct {
+	outputs map[string]*outputData
+}
+
+type outputData struct {
+	Value cty.Value
+}
+
+func (d dataSourceOutputs) readStateOutput(ctx context.Context, tfeClient *tfe.Client, orgName, wsName string) (*stateData, error) {
+	log.Printf("[DEBUG] Reading the Workspace %s in Organization %s", wsName, orgName)
+	ws, err := tfeClient.Workspaces.Read(ctx, orgName, wsName)
+	if err != nil {
+		return nil, fmt.Errorf("Error reading workspace: %v", err)
+	}
+
+	log.Printf("[DEBUG] Reading the current StateVersion for Workspace ID %s.", ws.ID)
+	sv, err := tfeClient.StateVersions.Current(ctx, ws.ID)
+	if err != nil {
+		if err == tfe.ErrResourceNotFound {
+			return nil, fmt.Errorf("Current remote state for workspace '%s' not found.", wsName)
+		}
+		return nil, fmt.Errorf("Could not read the current state for workspace '%s' : %v", wsName, err)
+	}
+
+	log.Printf("[DEBUG] Downloading State Version")
+	rawState, err := tfeClient.StateVersions.Download(ctx, sv.DownloadURL)
+	if err != nil {
+		return nil, fmt.Errorf("Error downloading state: %v", err)
+	}
```

Comment:
Whoops! I missed this earlier. We'll want the ability to be able to pull output values _without_ needing the entire state version; that's one of the clear advantages of using this over `terraform_remote_state`, as with TFC you can give [explicit access to outputs only](https://www.terraform.io/docs/cloud/users-teams-organizations/permissions.html#general-workspace-permissions).